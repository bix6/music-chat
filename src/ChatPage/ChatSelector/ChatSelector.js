import React from "react";

class ChatSelector extends React.Component {
  static defaultProps = {
    chatroomList: [],
  };

  // Create an option elem for each chatroom in the chatroom list
  createSelectOptions = () => {
    return this.props.chatroomList.map((chatroomObj, i) => {
      return (
        <option value={chatroomObj.name} key={i}>
          {chatroomObj.name}
        </option>
      );
    });
  };

  render() {
    // TODO is there a better way to handle this?
    let curChatroomDescription = null;
    this.props.chatroomList.forEach((chatroomObj) => {
      if (chatroomObj.name === this.props.currentChatroom) {
        curChatroomDescription = chatroomObj.description;
      }
    });

    return (
      <form className="chat-selector-form">
        <label htmlFor="chat-select">Chatroom: </label>
        <select
          name="chat-select"
          id="chat-select"
          value={this.props.currentChatroom}
          onChange={(e) => this.props.updateCurrentChatroom(e.target.value)}
        >
          {this.createSelectOptions()}
        </select>
        <button type="button" onClick={this.props.displayNewChatroom}>
            Create Chatroom
        </button>
        <p>{curChatroomDescription}</p>
      </form>
    );
  }
}

export default ChatSelector;
