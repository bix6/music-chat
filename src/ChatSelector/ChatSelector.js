import React from 'react';
import './ChatSelector.css';
import ChatContext from '../ChatContext';

class ChatSelector extends React.Component {
    static contextType = ChatContext;

    createSelectOptions = () => {
        const selectOptions = this.context.chatroomList.map((chatroomName, i) => {
            return <option value={ chatroomName } className="chat-select-option" key={ i }>{ chatroomName }</option>
        })
        return selectOptions;
    }

    render() {
        return (
            <div className="chat-selector">
                <label htmlFor="chat-select">Chat Room:</label>
                <select name="chat-select" id="chat-select" onChange= { e => this.context.updateChatroom(e.target.value) }>
                    { this.createSelectOptions() }
                </select>
            </div>
        )
    }
}

export default ChatSelector;