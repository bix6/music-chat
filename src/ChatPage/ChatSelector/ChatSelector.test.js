import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import ChatSelector from "./ChatSelector";

describe("ChatSelector Component", () => {
  it("Smoke Test: Renders Default", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <ChatSelector />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot Test: Default", () => {
    const wrapper = shallow(<ChatSelector />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
