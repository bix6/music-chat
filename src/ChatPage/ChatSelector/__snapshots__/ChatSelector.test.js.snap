// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatSelector Component Snapshot Test: Default 1`] = `
<form
  className="chat-selector-form"
>
  <label
    htmlFor="chat-select"
  >
    Chatroom: 
  </label>
  <select
    id="chat-select"
    name="chat-select"
    onChange={[Function]}
  />
  <button
    type="button"
  >
    Create Chatroom
  </button>
  <p />
</form>
`;
